gp-includes/strings.php

```diff
@@ -96,21 +96,23 @@ function gp_levenshtein( $str1, $str2, $length1, $length2 ) {
 		return 0;
 	}
 
-	$bytelength1 = strlen( $str1 );
-	$bytelength2 = strlen( $str2 );
+	if ( $length1 <= 255 && $length2 <= 255 ) {
+		$bytelength1 = strlen( $str1 );
+		$bytelength2 = strlen( $str2 );
 
-	if ( $bytelength1 === $length1 && $bytelength1 <= 255
-		 && $bytelength2 === $length2 && $bytelength2 <= 255 ) {
-		return levenshtein( $str1, $str2 );
+		if ( $bytelength1 === $length1 && $bytelength2 === $length2 ) {
+			return levenshtein( $str1, $str2 );
+		}
 	}
 
+	$chars1 = preg_split( '//u', $str1, -1, PREG_SPLIT_NO_EMPTY );
+	$chars2 = preg_split( '//u', $str2, -1, PREG_SPLIT_NO_EMPTY );
```

Comment:
PHP 7.4 has introduced `mb_str_split()` which we could use here instead. In my testing with a 81 chars long string and 100k iterations `preg_split` took 0.78s while `mb_str_split` only 0.16s.

If there's such an improvement I have no objections for bumping the required PHP version. See also #1357.

Reply:
That sounds like a better option here for sure! I'll update the PR with that when I get a moment to test it.

I would offer up a polyfill variant, but bumping the PHP requirement seems much more sane here, especially since PHP 7.3 has already fallen out of security support which is the GP requirement per #1357.